connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectExecutePlanHandler.scala

```diff
@@ -40,6 +40,8 @@ class SparkConnectExecutePlanHandler(responseObserver: StreamObserver[proto.Exec
         // Detached before execution finished.
         // TODO this doesn't happen yet without reattachable execution.
         responseObserver.onCompleted()
+      } else {
+        executeHolder.events.postClosed()
       }
     } finally {
       // TODO this will change with detachable execution.
```

Comment:
for now just put it in the finally, since the detached code path can't happen yet (and e.g. `executeHolder.join()` should also not be happening for detached).
this will change soon.